File: Session14/C++/TopologicalSort.cpp
https://practice.geeksforgeeks.org/problems/topological-sort/1/

    void findTopoSort(int node, vector<int> adj[], stack<int>& s, vector<int>& visited) {
        visited[node] = 1;
        for(int nbr:adj[node]) {
            if(!visited[nbr]) {
                findTopoSort(nbr, adj, s, visited);
            }
        }
        s.push(node);
    }
	public:
	//Function to return list containing vertices in Topological order. 
	vector<int> topoSort(int V, vector<int> adj[]) 
	{
	    // code here
	    stack<int> s;
	    vector<int> visited(V, 0);
	    
	    for(int i=0; i<V; i++) {
	        if(!visited[i]) {
	            findTopoSort(i, adj, s, visited);
	        }
	    }
	    
	    vector<int> res;
	    while(!s.empty()) {
	        res.push_back(s.top());
	        s.pop();
	    }
	    
	    return res;
	}

File: Session14/Java/TopologicalSort.java
https://practice.geeksforgeeks.org/problems/topological-sort/1/

class Solution
{
    static void findTopoSort(int node, int vis[], ArrayList<ArrayList<Integer>> adj, Stack<Integer> st) 
    {
        vis[node] = 1; 
        for(Integer nbr: adj.get(node)) {
            if(vis[nbr] == 0) {
                findTopoSort(nbr, vis, adj, st); 
            } 
        }
        st.push(node); 
    }
    
    //Function to return list containing vertices in Topological order. 
    static int[] topoSort(int V, ArrayList<ArrayList<Integer>> adj) 
    {
        // add your code here
        Stack<Integer> st = new Stack<Integer>(); 
        int vis[] = new int[V]; 
        
        for(int i=0; i<V; i++) {
            if(vis[i] == 0) {
                findTopoSort(i, vis, adj, st);
            }
        }
        
        int res[] = new int[V];
        int ind = 0; 
        while(!st.isEmpty()) {
            res[ind++] = st.pop();
        }

        return res; 
    }
}